src/index.jsx

```diff
@@ -36,53 +40,99 @@ export default class Script extends React.Component {
   constructor(props) {
     super(props);
     this.scriptLoaderId = `id${this.constructor.idCount++}`; // eslint-disable-line space-unary-ops, no-plusplus
+    this.pkID = `script-${shortid.generate()}`;
+    this.unmount = () => {
+      const { url, onUnload } = this.props;
+      const observers = this.constructor.scriptObservers[url];
+      const loaded = this.constructor.loadedScripts[url];
+      const errored = this.constructor.erroredScripts[url];
+
+      // If the component is waiting for the script to load, remove the
+      // component from the script's observers before unmounting the component.
+      if (observers) {
+        delete observers[this.scriptLoaderId];
+        // delete (this.constructor.scriptObservers)[url];
```

Comment:
I think It must not be commented, but your latest test fails if I uncomment this line